Service account tokens (#1045)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-cloud/dbt-cloud-api/service-tokens.md

---
title: "Service account tokens"
id: "service-tokens"
description: "Service account tokens help you define permissions for securing access to your dbt Cloud account and its projects."
---

## About service tokens

Service account tokens enable you to securely authenticate with the dbt Cloud API by assigning each token a narrow set of permissions that more precisely manages access to the API. While similar to [User API tokens](user-tokens), service account tokens belong to an account rather than a user.

You can use service account tokens for system-level integrations that do not run on behalf of any one user. Assign any role available in dbt Cloud to your service account token. For more information about roles, see "[Enterprise Permissions](docs/dbt-cloud/access-control/enterprise-permissions)".

## Generating service account tokens

In the Account Settings view of dbt Cloud, you can click on the Service Account tokens page and generate a new token. Create and save your token somewhere safe.

:::caution Note

You will not be able to view this token again after generating it, so store the token somewhere safe for later use.

:::
## Permissions for service account tokens

> You can assign service account tokens any role available in dbt Cloud. When you assign a role to a token, you will also be able to choose whether to grant that permissions to all projects in the account or to specific projects.

Reviewer: Can we replace the notion of this "role" with "permission" (or maybe "permission set") in this document? Eg.

> You can assign service account tokens any _permission set_ available in dbt Cloud. When you assign a _permission set_ to a token, you will also be able to choose whether to grant that permission set to all projects in the account or to specific projects.

Author: Addressed this in https://github.com/dbt-labs/docs.getdbt.com/pull/1045/commits/87e06baf1e765ee87f84594d9bc77bcf988b7210

> The following permission list summarizes each role. For more details about the permissions in these roles, see "[Enterprise permissions](/docs/dbt-cloud/access-control/enterprise-permissions)."

Reviewer: I wonder if we should decouple the list of permission sets from the list of "enterprise permissions". Non-Enterprise accounts can now use these permission sets when creating Service Tokens, so documenting them under "Enterprise Permissions" is kind of confusing!

Author: @drewbanin Who all can use these permission sets? Is it Teams and Enterprise accounts?

Reviewer: Will let @saimaddali chime in here with info about the implementation :)

Author: @saimaddali and @drewbanin I think I will address the Enterprise permissions page separately from this PR. I'd like to do it thoughtfully and not squeeze it in with this release 🥂 

Sai, maybe we can talk more about Drew's great point when we sync up this week?

Author: (I did make separate lists for service tokens so hopefully that will help)

Author: I will create a follow up issue for this suggestion!

---
title: "Service account tokens"
id: "service-tokens"
description: "Service account tokens help you define permissions for securing access to your dbt Cloud account and its projects."
---

## About service tokens

Service account tokens enable you to securely authenticate with the dbt Cloud API by assigning each token a narrow set of permissions that more precisely manages access to the API. While similar to [User API tokens](user-tokens), service account tokens belong to an account rather than a user.

You can use service account tokens for system-level integrations that do not run on behalf of any one user. Assign any permissions set available in dbt Cloud to your service account token.

:::info Enterprise and Team plan differences

> Enterprise plans can apply any permission sets available to service tokens. Team plans can apply three permissions sets to service tokens: Owner, Metadata, or Job Admin.

Author: @parose @saimaddali Should I add `Member` and `Read-only` here?

Reviewer 2: Yep, I think this should read:

>Team plans can apply the following permissions sets to service tokens: Account Admin, Member, Job Admin, Read-Only, and Metadata Only.

Author: Fixed in https://github.com/dbt-labs/docs.getdbt.com/pull/1045/commits/4c897e38fe49435a866d656afff5bccb7cdb417e

> Enterprise plans can apply any permission sets available to service tokens. Team plans can apply three permissions sets to service tokens: Owner, Metadata, or Job Admin.

Author: ```suggestion
Enterprise plans can apply any permission sets available to service tokens. Team plans can apply the following permissions sets to service tokens: Account Admin, Member, Job Admin, Read-Only, and Metadata Only.
```

> Enterprise plans can apply any permission sets available to service tokens. Team plans can apply the following permissions sets to service tokens: Owner, Member, Job Admin, Read-Only, and Metadata Only.

Author: ```suggestion
Enterprise plans can apply any permission sets available to service tokens. Team plans can apply the following permissions sets to service tokens: Account Admin, Member, Job Admin, Read-Only, and Metadata Only.
```

> Enterprise plans can apply any permission sets available to service tokens. Team plans can apply the following permissions sets to service tokens: Account Admin, Member, Job Admin, Read-Only, and Metadata Only.

Author: ```suggestion
Enterprise plans can apply any permission sets available to service tokens. Team plans can apply the following permissions sets to service tokens: Owner, Member, Job Admin, Read-Only, and Metadata Only.
```

Reviewer 2: `Account Admin` was correct here! That's how we're displaying the permission in the UI, even for tokens on Team plans, as before these changes we referred to them as Account Admin tokens on all plans

Author: Fixed back to "Account Admin"  in https://github.com/dbt-labs/docs.getdbt.com/pull/1045/commits/ef7f3e46e6ecad73b856227a17feac89d892091f

:::

For more on permissions sets, see "[Enterprise Permissions](docs/dbt-cloud/access-control/enterprise-permissions)."

## Generating service account tokens

In the Account Settings view of dbt Cloud, you can click on the Service Account tokens page and generate a new token. Create and save your token somewhere safe.

:::caution Note

You will not be able to view this token again after generating it, so store the token somewhere safe for later use.

:::

## Permissions for service account tokens

You can assign service account tokens any permission set available in dbt Cloud. When you assign a permission set to a token, you will also be able to choose whether to grant that permissions to all projects in the account or to specific projects.

### Service account tokens for Team plans

The following permissions can be assigned to a service account token on a Team plan.

> **Owner service token:**<br/>

Reviewer 2: I think lines 37 and 38 should still read "Account Admin", since that's how we're showing it in the UI (even for users on Team plans)

Author: Yes! Thanks for confirming that, Andrew! Changed in https://github.com/dbt-labs/docs.getdbt.com/pull/1045/commits/d65504ba624440633395fba716c2479cced2f2ab

> **Owner service token:**<br/>

Author: ```suggestion
**Account Admin service token:**<br/>
```

> **Account Admin service token:**<br/>

Author: ```suggestion
**Owner service token:**<br/>
```

> Account Admin service tokens have full `read + write` access to an account, so please use them with caution.  A Team plan refers to this permission set as an "Owner role."

Reviewer 2: Should we link to the Account Admin section from [this page](https://docs.getdbt.com/docs/dbt-cloud/access-control/enterprise-permissions#account-admin) here? Or did we remove that to avoid confusion between Team/Enterprise plans?

Author: I did remove it to avoid confusion. @saimaddali should I add it back?

Author: I added this link back in https://github.com/dbt-labs/docs.getdbt.com/pull/1045/commits/4e8e5e45fdcfefb4d52c5ac54e8ad30777a72d75

> Owner service tokens have full `read + write` access to an account, so please use them with caution.  A Team plan refers to this permission set as an "Owner role."

Author: ```suggestion
Account Admin service tokens have full `read + write` access to an account, so please use them with caution.  A Team plan refers to this permission set as an "Owner role."
```

> Account Admin service tokens have full `read + write` access to an account, so please use them with caution.  A Team plan refers to this permission set as an "Owner role."

Author: ```suggestion
Owner service tokens have full `read + write` access to an account, so please use them with caution.  A Team plan refers to this permission set as an "Owner role."
```

**Metadata Only service token:**<br/>
Metadata only service tokens can authorize requests to the metadata API.

**Job Admin service token:**<br/>

> Job admin service tokens can authorize requests for viewing, editing, and creating environments, triggering runs, and viewing historical runs.  

Reviewer 2: Similar question to above - should we link to the Job Admin section from the Enterprise Permissions page?

Author: added link back here https://github.com/dbt-labs/docs.getdbt.com/pull/1045/commits/4e8e5e45fdcfefb4d52c5ac54e8ad30777a72d75

**Member service token** <br/>

> Member service tokens can  authorize requests for viewing and editing resources, triggering runs, and inviting members to the account.

Reviewer 2: For Member and Read-Only, should we include a link to [this page](https://docs.getdbt.com/docs/dbt-cloud/access-control/self-service-permissions) and mention something like "Tokens assigned the Member permission set will have the same permissions as a Member user"?

Author: I will follow up later with a better explanation of read-only. The link didn't seem to clearly explain what they'd need to know for read-only.

> Member service tokens can  authorize requests for viewing and editing resources, triggering runs, and inviting members to the account.

Author: ```suggestion
Member service tokens can authorize requests for viewing and editing resources, triggering runs, and inviting members to the account. Tokens assigned the Member permission set will have the same permissions as a Member user. For more information about Member users, see "[Self-service permissions](/dbt-cloud/access-control/self-service-permissions)".
```
